generic_config_updater/patch_sorter.py

```diff
@@ -0,0 +1,965 @@
+        return self.config_wrapper.validate_config_db_config(simulated_config)
+
+# TODO: Add this validation YANG models instead
+class UniqueLanesMoveValidator:
+    def validate(self, move, diff):
+        simulated_config = move.apply(diff.current_config)
+
+        if "PORT" not in simulated_config:
+            return True
+
+        ports = simulated_config["PORT"]
+        existing = set()
+        for port in ports:
+            attrs = ports[port]
+            if "lanes" in attrs:
+                lanes_str = attrs["lanes"]
+                lanes = lanes_str.split(", ")
+                for lane in lanes:
+                    if lane in existing:
+                        return False
+                    existing.add(lane)
+        return True
+
+# TODO: Add 'create-only' extension to YANG models
+class CreateOnlyMoveValidator:
+    def __init__(self, path_addressing):
+        self.path_addressing=path_addressing
+
+    def validate(self, move, diff):
+        if move.op_type != OperationType.REPLACE:
+            return True
+
+        # The 'create-only' field needs to be common between current and simulated anyway but different.
+        # This means it is enough to just get the paths from current_config, paths that are not common can be ignored.
+        paths = self._get_create_only_paths(diff.current_config)
+        simulated_config = move.apply(diff.current_config)
+
+        for path in paths:
+            tokens = self.path_addressing.get_path_tokens(path)
+            if self._value_exist_but_different(tokens, diff.current_config, simulated_config):
+                return False
+
+        return True
+
+    def _get_create_only_paths(self, config):
+        if "PORT" not in config:
+            return
+
+        ports = config["PORT"]
+
+        for port in ports:
+            attrs = ports[port]
+            if "lanes" in attrs:
+                yield f"/PORT/{port}/lanes"
+
+    def _value_exist_but_different(self, tokens, current_config_ptr, simulated_config_ptr):
+        for token in tokens:
+            mod_token = int(token) if isinstance(current_config_ptr, list) else token
+
+            if mod_token not in current_config_ptr:
+                return False
+
+            if mod_token not in simulated_config_ptr:
+                return False
+
+            current_config_ptr=current_config_ptr[mod_token]
+            simulated_config_ptr=simulated_config_ptr[mod_token]
+
+        return current_config_ptr!=simulated_config_ptr
+
+class NoDependencyMoveValidator:
+    def __init__(self, path_addressing, config_wrapper):
+        self.path_addressing=path_addressing
+        self.config_wrapper=config_wrapper
+
+    def validate(self, move, diff):
+        operation_type = move.op_type
+        path = move.path
+
+        if operation_type == OperationType.ADD:
+            simulated_config = move.apply(diff.current_config)
+            if not self._validate_paths_config([path], simulated_config):
```

Comment:
Add a comment:
`For add operation, we check the simulated config has no dependencies between nodes under the added path`